modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/SignatureType.scala

```diff
@@ -0,0 +1,23 @@
+package org.tessellation.rosetta.domain
+
+import derevo.cats.{eqv, show}
+import derevo.derive
+import derevo.scalacheck.arbitrary
+import enumeratum.values.{StringCirceEnum, StringEnum, StringEnumEntry}
+
+@derive(arbitrary, eqv, show)
+sealed abstract class SignatureType(val value: String) extends StringEnumEntry
+
+object SignatureType extends StringEnum[SignatureType] with StringCirceEnum[SignatureType] {
+  val values = findValues
+
+  case object Ecdsa extends SignatureType("ecdsa")
+
+  case object EcdsaRecovery extends SignatureType("ecdsa_recovery")
+
+  case object Ed25519 extends SignatureType("ed25519")
+
+  case object Schnorr1 extends SignatureType("schnorr_1")
+
+  case object SchnorrPoseidon extends SignatureType("schnorr_poseidon")
```

Comment:
Our network only uses `ecdsa` right now. Let's remove the other ones until we need them.